### PR TITLE
[#4084] feat(bundle-catalog): shadow iceberg package in bundle catalog

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
shadow iceberg package in bundle catalog

### Why are the changes needed?
Fix: #4084 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
after this fix, spark it run pass after split Iceberg common logic

### CODE REVIEW

**Comment by FANNG1** [comment]
@yuqi1129 @diqiu50 @jerryshao , please help to review when you are free, thanks

File: catalogs/bundled-catalog/build.gradle.kts [hunk 1]
```diff
@@ -85,6 +85,7 @@ tasks.withType<ShadowJar>(ShadowJar::class.java) {
   exclude("NOTICE")
 
   minimize()
+  relocate("com.datastrato.gravitino.catalog.lakehouse.iceberg", "com.datastrato.gravitino.shaded.bundled.catalog.lakehouse.iceberg")
```
Comment (yuqi1129) [inline catalogs/bundled-catalog/build.gradle.kts:88]: I'm not very sure why only to relocate the package of Iceberg catalog, are there any similar issues for Hive and Hadoop catalogs?

Comment (FANNG1) [reply catalogs/bundled-catalog/build.gradle.kts:88]: no, it had problem when the classes referred by `XXXPropertiesMetadata` is moved from the module

Comment (yuqi1129) [reply catalogs/bundled-catalog/build.gradle.kts:88]: I see. Relocating Iceberg catalog-related code is not so elegant but I can't come up with a better choice.

**Comment by FANNG1** [comment]
@jerryshao  do you have time to review this PR?

File: catalogs/bundled-catalog/build.gradle.kts (hunk 1 above)
Comment (jerryshao) [reply catalogs/bundled-catalog/build.gradle.kts:88]: I think this is not what bundled-catalog originally proposed to do, it is weird to shade Iceberg catalog package in bundled-catalog.

With this, there may have two Gravitino Iceberg package, one shaded and one not, will this lead to problem easily?

Comment (FANNG1) [reply catalogs/bundled-catalog/build.gradle.kts:88]: bundle catalog is used to provide the constants shared by connectors and Gravitino catalogs. The current implementation used the constant from `XXXPropertiesMetadata` and introduced the referred classes in `bundle-catalog`,  seems not proper, How about defining constant in `bundle catalog` and `XXXPropertiesMetadata` rely on the constants?

Comment (jerryshao) [reply catalogs/bundled-catalog/build.gradle.kts:88]: Yeah, that's better than the current solution.

Comment (FANNG1) [reply catalogs/bundled-catalog/build.gradle.kts:88]: Propose #4122 to use constants defined in `bundled catalog`, @jerryshao @yuqi1129 please help to review, I'll close this PR

**Pull request closed without merging**